### PR TITLE
add an authentication_required option

### PR DESCRIPTION
**Pull request by jwkaltz**
See https://jira.camptocamp.com/browse/GEO-1749

### CODE REVIEW

File: c2cgeoportal/views/mapserverproxy.py [hunk 1]
```diff
@@ -51,6 +52,11 @@ def __init__(self, request):
     @view_config(route_name="mapserverproxy")
     def proxy(self):
 
+        if self.user is None:
+            if "authentication_required" in self.request.params:
```
Comment (sbrunner) [inline c2cgeoportal/views/mapserverproxy.py]:
```suggestion
        if self.user is None and "authentication_required" in self.request.params:
```
In one line

File: c2cgeoportal/views/mapserverproxy.py [hunk 2]
```diff
@@ -51,6 +52,11 @@ def __init__(self, request):
...
+                log.debug("proxy() detected authentication_required")
+                raise HTTPUnauthorized(headers={"WWW-Authenticate": 'Basic realm="Access to all layers"'})
```
Comment (sbrunner) [inline c2cgeoportal/views/mapserverproxy.py]:
```suggestion
                raise HTTPUnauthorized(headers={"WWW-Authenticate": 'Basic realm="Access to restricted layers"'})
```

File: doc/integrator/security.rst [hunk 3]
```diff
@@ -11,6 +11,27 @@ the WMS GetCapability when accessing the Mapserver proxy (mapserverproxy).
...
+If your WMS contains private layers and you wish to force a client to authenticate
+in order to always obtain all layers (including private layers), you can force
+authentication by adding the parameter ``authentication_required`` to the WMS URL.
+This setting may be necessary for the good operation of some clients such as ArcMap.
+
+For example, if your WMS is accessible as
+
+.. code:: html
+
+   https://yourserver/yourinstance/wsgi/mapserv_proxy
```
Comment (sbrunner) [inline doc/integrator/security.rst]:
```suggestion
   https://<yourserver>/<yourinstance>/wsgi/mapserv_proxy
```

File: doc/integrator/security.rst [hunk 4]
```diff
@@ -11,6 +11,27 @@ the WMS GetCapability when accessing the Mapserver proxy (mapserverproxy).
...
+
+then you can use the following URL to force authentication:
+
+.. code:: html
+
+   https://yourserver/yourinstance/wsgi/mapserv_proxy?authentication_required=true
```
Comment (sbrunner) [inline doc/integrator/security.rst]:
```suggestion
   https://<yourserver>/<yourinstance>/wsgi/mapserv_proxy?authentication_required=true
```

**Comment by jwkaltz** [comment]
@sbrunner I added a test, for better coverage - OK for you?
What about Jenkins, can we trigger it somehow?

**Comment by sbrunner** [comment]
If Jenkins pass it's OK for me :-)

**Pull request merged**